crates/circuit/src/circuit_instruction.rs

```diff
@@ -618,6 +622,32 @@ impl<'a, 'py> FromPyObject<'a, 'py> for OperationFromPython {
                     });
                 };
             }
+        } else if ob_name == "pauli_product_measurement" {
+            let z = ob
+                .getattr(intern!(py, "_pauli_z"))?
+                .extract::<PyReadonlyArray1<bool>>()?
+                .as_slice()?
+                .to_vec();
+
+            let x = ob
+                .getattr(intern!(py, "_pauli_x"))?
+                .extract::<PyReadonlyArray1<bool>>()?
+                .as_slice()?
+                .to_vec();
+
+            let phase = ob.getattr(intern!(py, "_pauli_phase"))?.extract::<u8>()?;
+
+            let pauli_product_measurement = Box::new(PauliProductMeasurement {
+                z: z.to_owned(),
+                x: x.to_owned(),
+                neg: phase == 2,
```

Comment:
It might be worth adding a clarifying comment here that explains that `phase` can only be 0 or 2 from input validation